runtime/gc_vlhgc/AllocationContextBalanced.cpp

```diff
@@ -391,18 +391,25 @@ MM_AllocationContextBalanced::lockedAllocateArrayletLeaf(MM_EnvironmentBase *env
 		 * In future, allocations should remember (somewhere in Off-heap meta structures) how many regions came from each AC
 		 * \and release exact same number back to each AC.
 		 */
-		MM_AllocationContextTarok *commonContext = (MM_AllocationContextTarok *)env->getCommonAllocationContext();
-		if (this != commonContext) {
+		MM_AllocationContextTarok *context = leafAllocateData->_owningContext;
+		if (NULL != leafAllocateData->_originalOwningContext) {
+			context = leafAllocateData->_originalOwningContext;
+		}
+
+		Assert_MM_true(NULL != context);
+
+		allocateDescription->setAllocationContext(context);
+		if (this != context) {
 			/* The common allocation context is always an instance of AllocationContextBalanced */
-			((MM_AllocationContextBalanced *)commonContext)->lockCommon();
+			((MM_AllocationContextBalanced *)context)->lockCommon();
```

Comment:
At this point we hold locks of 2 different ACs. There should be no scenario that another thread attempts to lock these same locks in opposite order, since it would not able to successfully steal from us (our AC is already depleted).